File: csharp/decorator_practice/decorator_practice/test/AdminUserTest.cs
using System;
using System.Collections.Generic;
using System.Text;
using decorator_practice.main.decorator;
using decorator_practice.main.users;
using Moq;
using NUnit.Framework;

namespace decorator_practice.test
{
    [TestFixture]
    class AdminUserTest
    {
        public Mock<User> user;
        public AdminUser au;

        [SetUp]
        public void SetUp()
        {
            au = new AdminUser();
            user = new Mock<User>(au);
        }

        [TestCase]
        public void NameTest()
        {
            Assert.AreEqual("Admin", user.Object.getName());
        }
    }
}

File: java/src/test/AdminUserTest.java
package mx.iteso;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;
import static org.junit.Assert.Equals;
import static org.junit.Assert.assertTrue;
import static org.junit.Before;
import static org.junit.Test;

/**
 * Admin user test class.
 */
public class AdminUserTest {
    private User user;

    /**
     * Set up method.
     */
    @Before
    public void SetUp(){
        user = mock((AdminUser.class));
    }    
    /**
     * Get name test.
     */
    @Test
    public void NameTest(){
        when(user.getName()).thenReturn("Admin");
    }
}
